lib/semian/net_http.rb

```diff
@@ -0,0 +1,94 @@
+require 'semian'
+require 'semian/adapter'
+
+module Net
+  ProtocolError.include(::Semian::AdapterError)
+
+  class SemianError < ::Net::ProtocolError
+    def initialize(semian_identifier, *args)
+      super(*args)
+      @semian_identifier = semian_identifier
+    end
+  end
+
+  ResourceBusyError = Class.new(SemianError)
+  CircuitOpenError = Class.new(SemianError)
+end
+
+module Semian
+  module NetHTTP
+    include Semian::Adapter
+
+    ResourceBusyError = ::Net::ResourceBusyError
+    CircuitOpenError = ::Net::CircuitOpenError
+
+    def semian_identifier
+      @configuration ||= Semian::NetHTTP.retrieve_semian_configuration_by_host_port(address, port)
```

Comment:
@Sirupsen, I can't imagine a situation where a library will use a `Net::HTTP` connection through an instance of `Net::HTTP` object, and then somehow have hot-swapped closures, and still use the same `Net::HTTP` instance. If you think it's a valid concern, then I'll just remove the memoization